### PR TITLE
Sptr 4/parameter as lists

### PR DESCRIPTION
**Pull request by theseanything**
Add functionality to format a yaml list of parameters in a stack config as an AWS compatible comma separated list.

### CODE REVIEW

File: docs/stack_config.rst [hunk 1]
```diff
@@ -39,6 +39,44 @@ A list of arbitrary shell or python commands or scripts to run. Find out more in
 
 A dictionary of key-value pairs to be supplied to a CloudFormation or Troposphere template as parameters. The keys must match up with the name of the parameter, and the value must be of the type as defined in the template. Note that Boto3 throws an exception if parameters are supplied to a template that are not required by that template. Resolvers can be used to add functionality to this key. Find out more in the `Resolvers`_ section.
 
+Parameters can be specified as a single or list of values/resolvers. Lists of values/resolvers will be formatted into an AWS compatible comma separated string e.g. "value1,value2,value3". Lists can contain a mixture of values and resolvers.
```
Comment (jamesroutley) [inline docs/stack_config.rst]: Could we make it:

Parameters can be specified as a single **value/resolver** or **a** list of values/resolvers.

It's slightly ambiguous as is: 

Parameters can be specified as a single (?) or parameters can be supplied as a list of values/resolvers

Comment (theseanything) [inline docs/stack_config.rst]: Yup agree. See change

File: sceptre/stack.py [hunk 2]
```diff
@@ -651,8 +651,18 @@ def _format_parameters(self, parameters):
         :returns: A list of the formatted parameters.
         :rtype: list
         """
-        return [{"ParameterKey": name, "ParameterValue": value}
-                for name, value in parameters.items() if value is not None]
+        formatted_parameters = []
+        for name, value in parameters.items():
+            if value is None:
```
Comment (jamesroutley) [inline sceptre/stack.py]: Could we have a unittest for this guy?

File: docs/stack_config.rst [hunk 3]
```diff
@@ -39,6 +39,44 @@ A list of arbitrary shell or python commands or scripts to run. Find out more in
...
+
+.. code-block:: yaml
+
+  parameters:
+    <parameter1_name>: "value"
+    <parameter2_name>: !<resolver_name> <resolver_value>
+    <parameter3_name>:
+      - "value1"
+      - "value2"
+      - "value3"
```
Comment (jamesroutley) [inline docs/stack_config.rst]: Could we just have two items in this list? I think it gets the point across and is slightly terser.

File: docs/stack_config.rst [hunk 4]
```diff
@@ -39,6 +39,44 @@ A list of arbitrary shell or python commands or scripts to run. Find out more in
...
+    <parameter4_name>:
+      - !<resolver_name> <resolver_value>
+      - !<resolver_name> <resolver_value>
+      - !<resolver_name> <resolver_value>
```
Comment (jamesroutley) [inline docs/stack_config.rst]: And here

File: docs/stack_config.rst [hunk 5]
```diff
@@ -39,6 +39,44 @@ A list of arbitrary shell or python commands or scripts to run. Find out more in
...
+    <parameter5_name>:
+      - !<resolver_name> <resolver_value>
+      - "value1"
+      - !<resolver_name> <resolver_value>
```
Comment (jamesroutley) [inline docs/stack_config.rst]: And here

File: docs/stack_config.rst [hunk 6]
```diff
@@ -39,6 +39,44 @@ A list of arbitrary shell or python commands or scripts to run. Find out more in
...
+
+.. code-block:: yaml
+
+    parameters:
+      database_username: "mydbuser"
+      database_password: !environment_variable DATABASE_PASSWORD
+      subnet_ids:
+        - "subnet-12345678"
+        - "subnet-87654321"
+        - "subnet-11111111"
```
Comment (jamesroutley) [inline docs/stack_config.rst]: And here

File: docs/stack_config.rst [hunk 7]
```diff
@@ -39,6 +39,44 @@ A list of arbitrary shell or python commands or scripts to run. Find out more in
...
+      security_group_ids:
+        - "sg-12345678"
+        - !stack_output security-groups::BaseSecurityGroupId
+        - !file_contents /file/with/security_group_id.txt
```
Comment (jamesroutley) [inline docs/stack_config.rst]: I think there's value in having all three here - good to show that different resolvers can be used together.

File: sceptre/stack.py (hunk 2 above)
Comment (theseanything) [inline sceptre/stack.py]: Added.

**Pull request merged**